src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountSupervisor.kt

```diff
@@ -1296,6 +1299,91 @@ internal class SubaccountSupervisor(
+     */
+    private fun reclaimUnutilizedFundsFromChildSubaccounts() {
+        val subaccounts = stateMachine.state?.account?.subaccounts ?: return
+
+        val subaccountQuoteBalanceMap = subaccounts.mapValues { subaccount ->
+            // If the subaccount is the parentSubaccount, skip
+            if (subaccount.value.subaccountNumber == subaccountNumber) {
+                return@mapValues 0.0
+            }
+
+            val openPositions = subaccount.value.openPositions
+            val openOrders = subaccount.value.orders?.filter { order ->
+                val status = helper.parser.asString(order.status)
+                status == "OPEN"
+            }
+            val quoteBalance = subaccount.value.quoteBalance?.current ?: 0.0
+
+            // Only return a quoteBalance if the subaccount has no open positions or orders
+            if (openPositions.isNullOrEmpty() && openOrders.isNullOrEmpty() && quoteBalance > 0.0) {
+                quoteBalance
+            } else {
+                0.0
+            }
+        }.filter {
+            it.value > 0.0
+        }
+
+        val transferPayloadStrings = iMutableListOf<String>()
+
+        subaccountQuoteBalanceMap.forEach {
+            val childSubaccountNumber = it.key.toInt()
+            val amountToTransfer = it.value.toString()
+
+            val transferPayload = HumanReadableSubaccountTransferPayload(
+                childSubaccountNumber,
+                amountToTransfer,
+                accountAddress,
+                subaccountNumber,
+            )
+
+            val transferPayloadString = Json.encodeToString(transferPayload)
+            transferPayloadStrings.add(transferPayloadString)
+        }
+
+        recursivelyReclaimChildSubaccountFunds(transferPayloadStrings)
+    }
+
+    private fun recursivelyReclaimChildSubaccountFunds(transferPayloadStrings: MutableList<String>) {
+        if (transferPayloadStrings.isNotEmpty()) {
+            val transferPayloadString = transferPayloadStrings.removeAt(0)
+            helper.transaction(TransactionType.SubaccountTransfer, transferPayloadString) { response ->
+                val error = parseTransactionResponse(response)
+                if (error != null) {
+                    emitError(error)
+                } else {
+                    recursivelyReclaimChildSubaccountFunds(transferPayloadStrings)
+                }
+            }
+        }
+    }
+
+    private var reclaimUnutilizedFundsTimer: LocalTimerProtocol? = null
+        set(value) {
+            if (field !== value) {
+                field?.cancel()
+                field = value
+            }
+        }
+
+    private fun pollReclaimUnutilizedFunds() {
+        reclaimUnutilizedFundsTimer = null
+        helper.ioImplementations.threading?.async(ThreadingType.main) {
+            this.reclaimUnutilizedFundsTimer = helper.ioImplementations.timer?.schedule(
+                (10.seconds).inWholeSeconds.toDouble(),
+                null,
```

Comment:
can we use this `repeat` duration instead of recursively calling this function?

Reply:
Current recursive implementation waits for a successful callback reponse for each transfer and goes through the entire list of `childSubaccounts` then creates a fresh 10 sec timer before repeating. I don't think I can achieve the same effect with `repeat`.